test/cases/neuron/manifests/multi-node-test-neuron.yaml

```
@@ -24,82 +24,35 @@ spec:
                   fieldPath: status.podIP
             command:
             - /bin/sh
+            args:
             - -c
-            - |
-                echo password | sudo -S service ssh start
-
-                # Read the IPs from the hosts.txt file
-                WORKER_0_IP=$(sed -n '1p' /mnt/data/hosts.txt | tr -d '\r' | tr -d '\n' )
-                WORKER_1_IP=$(sed -n '2p' /mnt/data/hosts.txt | tr -d '\r' | tr -d '\n' )
+            - | 
+                WORKER_0_IP=$(getent hosts multi-node-nccom-test-worker-0.multi-node-nccom-test | awk '{print $1}')
+                WORKER_IPS=$WORKER_0_IP
+                for i in $(seq 1 $(({{.WorkerNodeCount}} - 1))); do
+                  WORKER_IPS="$WORKER_IPS $(getent hosts multi-node-nccom-test-worker-$i.multi-node-nccom-test | awk '{print $1}')"
+                done
```

could we just combine these?

```suggestion
                for i in $(seq 0 $(({{.WorkerNodeCount}} - 1))); do
                  WORKER_IPS="$WORKER_IPS $(getent hosts multi-node-nccom-test-worker-$i.multi-node-nccom-test | awk '{print $1}')"
                done
```

---

👍 sounds good
one thing here was we need the first worker to set an env var below
```
export NEURON_RT_ROOT_COMM_ID=$WORKER_0_IP:63182
```
but we can just make this an actual array and index it

---

Did this in [ec208b7](https://github.com/aws/aws-k8s-tester/pull/577/commits/ec208b7b9bceb55aae012bf5bba5ae33823e9ac7)